lib/langs/python.rb

```diff
@@ -0,0 +1,66 @@
+require 'rubygems'
+require 'json'
+require 'http'
+
+class PythonLang
+  COMPARATORS = />=|>|<=|<|==/
+  VERSION_NUM = /\d[\.\w]*/
+  SUFFIX_OPTION = /\s*(\[.*\])?/
+  NAME = /[^,]+/
+  REGEX = /^(#{NAME})\s*(#{COMPARATORS})\s*(#{VERSION_NUM})(\s*,\s*(#{COMPARATORS})\s*(#{VERSION_NUM}))?#{SUFFIX_OPTION}$/
+
+  API_URL = 'https://pypi.org/pypi/'
+
+  # Defaults for projects in this language
+  def self.locations(name)
+    ['requirements.txt']
+  end
+
+  # Find the latest versions of gems in this gemspec
+  #
+  # content - String contents of the gemspec
+  #
+  # Returns an Array of tuples of each dependency and its latest version: [[<Bundler::Dependency>, <Gem::Version>]]
+  def self.parse_file(name, content)
+    return nil unless content
+    if name =~ /\.txt$/
+      self.parse_requirements(content)
+    elsif name == 'Pipfile'
+      self.parse_pipfile(content)
+    end
+  end
+
+  def self.parse_requirements(content)
+    results = []
+    content.each_line do |line|
+      next if line =~ /^\s*-e/ # ignore 'editable' dependencies
+      if line =~ COMPARATORS
+        match = line.match(REGEX)
+        return nil unless match
```

Comment:
Currently bailing entirely when unable to parse a single line. Does that make sense, or better to just do `next`?

Reply:
Either skip the line, or mark it as `:unknown`, but I prefer to keep going.